packages/config/types/build.d.ts

```diff
@@ -18,9 +18,17 @@ import { TerserPluginOptions } from 'terser-webpack-plugin'
 
 type NuxtConfigurationLoaders = any // TBD
 
+interface NuxtBabelPresetEnv {
+  isServer: boolean;
+}
+
+interface NuxtBabelOptions extends Pick<TransformOptions, Exclude<keyof TransformOptions, 'presets'>> {
+  presets?: ((env: NuxtBabelPresetEnv, presets: [string, object | null]) => PluginItem[] | void) | PluginItem[] | null
```

Comment:
Is there really a case that we accept functional `presets` returning `void` ? Should it not be always an array ?

Assuming https://github.com/nuxt/nuxt.js/blob/dev/packages/webpack/src/config/base.js#L81 & https://github.com/nuxt/nuxt.js/blob/dev/packages/webpack/src/config/base.js#L89, 
`presets` function parameter should be named `defaultPreset` instead and seems to fit `PluginItem` type.

```suggestion
  presets?: ((env: NuxtBabelPresetEnv, defaultPreset: PluginItem) => PluginItem[]) | PluginItem[] | null
```

Reply:
`presets` can modify the second argument and return nothing.

`presets: [string, object | null]` is fixed strcture, so it shouldn't 100% follow the structure of PluginItem

Reply:
Alright all good !

Reply:
Arg name is changed, thanks